### PR TITLE
add new 'rpc' strategy for gas price parameter

### PR DESCRIPTION
**Pull request by weilbith**
Second try after I accidentally closed #7509 ...

Extends the `--gas-price` parameter with a new gas strategy that queries the
connected Ethereum RPC node for the current price. This is meant to be used
especially for Arbitrum networks, as the other strategies do not work there. The
gas system works slightly different for such chains.

Furthermore there is no more a default parsing result. This means that the
command-line option still has a default, but the parsing function is strictly
evaluating the passed value.

### CODE REVIEW

**Comment by weilbith** [comment]
And again:
Please note that unfortunately must ignore the linting errors here. :expressionless:

**Review by palango — APPROVED** [review]

File: raiden/utils/cli.py [hunk 1]
```diff
@@ -371,10 +372,14 @@ def fixed_gas_price_strategy(_web3, _transaction_params):
...
-            if gas_price_string == "fast":
+            if gas_price_string == "normal":
+                return fast_gas_price_strategy
+            elif gas_price_string == "fast":
                 return faster_gas_price_strategy
+            elif gas_price_string == "rpc":
+                return rpc_gas_price_strategy
             else:
-                return fast_gas_price_strategy
+                raise ConfigurationError(f"Unknown gas price (strategy): {gas_price_string}")
```
Comment (netcriptus) [inline raiden/utils/cli.py:384]: Using a dictionary would make this more maintainable, and easier to read:

```python
gas_strategies = {
    "normal": fast_gas_price_strategy,
    "fast": faster_gas_price_strategy,
    "rpc": rpc_gas_price_strategy
}

if gas_price_string not in gas_strategies:
    raise ConfigurationError(f"Unknown gas price (strategy): {gas_price_string}")

return gas_strategies[gas_price_string]
```

Comment (weilbith) [reply raiden/utils/cli.py:384]: Good idea! I'll take that.

Comment (weilbith) [reply raiden/utils/cli.py:384]: I'll just amend the commit as it is a single one and I don't see much sense to push a fixup commit here that then needs to be rebased. I think the commit is small enough that it is easy for everyone to review the changes due to this conversation.

Comment (weilbith) [reply raiden/utils/cli.py:384]: ✔️

File: raiden/utils/cli.py [hunk 2]
```diff
@@ -370,11 +371,20 @@ def fixed_gas_price_strategy(_web3, _transaction_params):
...
             gas_price_string = super().convert(value, param, ctx)
-            if gas_price_string == "fast":
-                return faster_gas_price_strategy
+            
+            if gas_price_string not in gas_strategies:
+                raise ConfigurationError(f"Unknown gas price (strategy): {gas_price_string}")
+
             else:
-                return fast_gas_price_strategy
+                return gas_strategies[gas_price_string]
```
Comment (netcriptus) [inline raiden/utils/cli.py]: Looking better, but this `else` is redundant now, we could do without it.

Comment (weilbith) [reply raiden/utils/cli.py]: I kinda like a good if-else block. But I don't have a too strong opinion for such a case here. So I'll follow your recommendation.

Comment (weilbith) [reply raiden/utils/cli.py]: Same commiting-strategy as [explained above](https://github.com/raiden-network/raiden/pull/7510#discussion_r816673545)

Comment (weilbith) [reply raiden/utils/cli.py]: ✔️

**Review by netcriptus — APPROVED** [review]

**Pull request merged**